cypher/cypher-docs/src/docs/dev/query-tuning.asciidoc

```diff
@@ -57,6 +89,30 @@ When Cypher is building execution plans, it looks at the schema to see if it can
 These index decisions are only valid until the schema changes, so adding or removing indexes leads to the execution plan cache being flushed.
 
 
+[[cypher-preparser-runtime]]
+=== Cypher runtime
+
+Using the execution plan, the query is executed -- and records returned -- by the query engine, or _runtime_.
+Depending on whether Neo4j Enterprise Edition or Neo4j Community Edition is used, there are two different runtimes available:
+
+Interpreted::
+In this runtime, the operators in the execution plan are chained together in a tree, where each non-leaf operator feeds from one or two child operators.
+The tree thus comprises nested iterators, and the records are streamed in a pipelined manner from the top iterator, which pulls from the next iterator and so on.
+
+Compiled::
+Algorithms are employed to group intelligently the operators in the execution plan in order to generate new combinations and orders of execution which are optimised for performance and memory usage.
+While this should lead to superior performance in most cases, it is still under development.
+
+[NOTE]
+The compiled runtime is only available in Neo4j Enterprise Edition.
+
+[options="header"]
+|===
+| Preparser option | Description | Default?
+| `runtime=interpreted` | This will force the query to use the interpreted runtime.  | This is the default option for some queries in Enterprise Edition, and the only option for all queries in Community Edition (so it is not necessary to use this option in Community Edition).
+| `runtime=compiled` | This will force the query to use the compiled runtime. | This is the default option for some queries in Enterprise Edition only. Setting this option for a query that does not support the compiled runtime will cause the query to fall back to using the interpreted runtime.
```

Comment:
The word 'force' seems to conflict with the concept of falling back. These two sections could be written differently. For 'runtime=interpreted' we could say that this will skip the consideration of the compiled runtime, and simply use the default interpreted. While 'runtime=compiled' will cause the execution engine to use the compiled runtime if it supports the query and fall back to the interpreted runtime if it does not.

Reply:
👍